File: .claude/fetch_page2.py
#!/usr/bin/env python3
import json
import os
import urllib.request
import ssl

# Load .env.local
env = {}
with open('../.env.local', 'r') as f:
    for line in f:
        if '=' in line:
            k, v = line.strip().split('=', 1)
            env[k.strip()] = v.strip()

url = env['NEXT_PUBLIC_SUPABASE_URL'].rstrip('/') + '/rest/v1/document_extractions'
params = '?id=eq.625fb21c-2755-4c58-b92d-a74beaf2c5fa&select=data'
full_url = url + params

headers = {
    'Authorization': f"Bearer {env['SUPABASE_SERVICE_ROLE_KEY']}",
    'apikey': env['SUPABASE_SERVICE_ROLE_KEY'],
    'Content-Type': 'application/json'
}

try:
    ctx = ssl.create_default_context()
    ctx.check_hostname = False
    ctx.verify_mode = ssl.CERT_NONE
    
    req = urllib.request.Request(full_url, headers=headers)
    with urllib.request.urlopen(req, timeout=10, context=ctx) as response:
        data = json.loads(response.read().decode())
        
    if not data or len(data) == 0:
        print("No extraction found")
        exit(1)
    
    extraction_data = data[0].get('data', {})
    
    # Get page 2
    page2_text = None
    
    if extraction_data.get('content_layers_v1', {}).get('pdf', {}).get('text', {}).get('pages'):
        pages = extraction_data['content_layers_v1']['pdf']['text']['pages']
        for page in pages:
            if page.get('page_number') == 2:
                page2_text = page.get('text', '')
                break
    
    if not page2_text and extraction_data.get('extraction', {}).get('evidence_v1', {}).get('page_text'):
        pages = extraction_data['extraction']['evidence_v1']['page_text']
        for page in pages:
            if page.get('page_number') == 2:
                page2_text = page.get('text', '')
                break
    
    if page2_text:
        print("=== PAGE 2 TEXT ===\n")
        print(page2_text)
        print("\n=== SEARCH RESULTS ===")
        
        import re
        tests = [
            ('90 days', r'90\s+days'),
            ('ninety days', r'ninety\s+days'),
            ('fully executed', r'fully\s+executed'),
            ('from the date', r'from\s+the\s+date'),
            ('of execution', r'of\s+execution'),
        ]
        
        text_lower = page2_text.lower()
        for name, pattern in tests:
            match = re.search(pattern, text_lower, re.IGNORECASE)
            if match:
                print(f"✓ {name}: FOUND at position {match.start()}")
                start = max(0, match.start() - 50)
                end = min(len(page2_text), match.end() + 50)
                print(f"  Context: ...{page2_text[start:end]}...")
            else:
                print(f"✗ {name}: NOT FOUND")
    else:
        print("Page 2 text not found in extraction")
        
except Exception as e:
    print(f"Error: {e}")
    import traceback
    traceback.print_exc()

File: .claude/fetch-page2.js
const fs = require('fs');
const path = require('path');

// Load env
const envPath = path.join(__dirname, '..', '.env.local');
const envContent = fs.readFileSync(envPath, 'utf-8');
const env = {};
envContent.split('\n').forEach(line => {
  const [key, value] = line.split('=');
  if (key?.trim() && value?.trim()) {
    env[key.trim()] = value.trim();
  }
});

const { createClient } = require('@supabase/supabase-js');

async function fetch() {
  const admin = createClient(env.NEXT_PUBLIC_SUPABASE_URL, env.SUPABASE_SERVICE_ROLE_KEY);
  
  const { data, error } = await admin
    .from('document_extractions')
    .select('data')
    .eq('id', '625fb21c-2755-4c58-b92d-a74beaf2c5fa')
    .single();

  if (error) {
    fs.writeFileSync('/tmp/page2-error.txt', `Error: ${error.message}`);
    process.exit(1);
  }

  const extractionData = data?.data || {};
  
  let page2Text = null;
  
  if (extractionData.content_layers_v1?.pdf?.text?.pages) {
    const page2 = extractionData.content_layers_v1.pdf.text.pages.find(p => p.page_number === 2);
    if (page2) {
      page2Text = (page2.text || '') + (page2.plain_text_blocks?.map(b => b.text || '').join('\n') || '');
    }
  }

  if (!page2Text && extractionData.extraction?.evidence_v1?.page_text) {
    const page2 = extractionData.extraction.evidence_v1.page_text.find(p => p.page_number === 2);
    if (page2) {
      page2Text = page2.text || '';
    }
  }

  fs.writeFileSync('/tmp/page2.txt', page2Text || 'NOT FOUND');
  console.log('Page 2 written to /tmp/page2.txt');
}

fetch().catch(err => {
  fs.writeFileSync('/tmp/page2-error.txt', err.message);
  process.exit(1);
});
